src/lib.rs

```diff
@@ -244,7 +244,7 @@ pub mod lib {
         /// # Arguments
         ///
         /// * `machine` - A mutable reference to the [`Machine`] to load the module into.
-        /// * `module_name` (Optional) - A pointer to a null-terminated UTF-8 string representing 
+        /// * `module_name` - A pointer to a null-terminated UTF-8 string representing
```

Comment:
You can make the module optional by interpreting a null pointer as a `None` if you like. `Option<*const T>`, like you did earlier, doesn't have a well defined layout and so _is not safe_ for use in FFI code[^1]. But, as `CStr` needs a non-null pointer, you can treat the null case as an alternative.

[^1]: On the other hand, `Option<&T>` has a well defined layout (guaranteed to be equivalent to `*const T`), and so is . This is due to the fact that references `&T` are non-nullable, so `Option<&T>` can use the null value to represent `None`.

Reply:
`Option<NonNull<T>>` would be ABI compatible with `*const T` (technically `*mut T`)

[`NonNull` docs](https://doc.rust-lang.org/std/ptr/struct.NonNull.html)
> Unlike *mut T, the pointer must always be non-null, even if the pointer is never dereferenced. This is so that enums may use this forbidden value as a discriminant – Option<NonNull<T>> has the same size as *mut T. However the pointer may still dangle if it isn’t dereferenced.